agentic/src/nodes/base.ts

```diff
@@ -263,68 +269,42 @@ export abstract class BaseNode extends KaiWorkflowEventEmitter {
         data: new AIMessageChunk(buffer),
       });
     }
-    if (response && !this.modelInfo.toolsSupported) {
+    if (response && !this.modelProvider.toolCallsSupported()) {
       response.tool_calls = toolCalls;
     }
     return response;
   }
 
-  private getRunnableWithTools(
+  private getInputWithTools(
     input: BaseLanguageModelInput,
-    enableTools?: boolean,
-    toolsSelectors?: string[],
-  ): {
-    inputWithTools: BaseLanguageModelInput;
-    runnable: Runnable<BaseLanguageModelInput, AIMessageChunk, BaseChatModelCallOptions>;
```

Comment:
the runnable is now managed internally by the model provider, agent doesn't need to worry about that